### ISSUE TITLE
[Feature Request] Line of Sight and/or Hit-Box Should Be Dependent on Height/Width

### ISSUE THREAD

**Issue by TheMCSpacePro** [Luke100000/minecraft-comes-alive#764]
**Is your feature request related to a problem? Please describe.**
Something I noticed is that no matter what the player's height/width is set to in the character creator, neither the player's hit-box nor line of sight changes in relation to the height/width. For the line of sight, this makes no since due to the fact that relative height of player's eyes from the ground changes based on height. For the player's hit-box, a sword strike right next to a super-skinny or super-short person (not touching them directly) shouldn't count as a hit, but does since it lands inside of a normal sized hit-box that's not proportional to the height and width of the player.

**Describe the solution you'd like**
Simple: two options in the config, one for the line of sight shifting up or down depending on the player's height, and another for the hit-box shrinking or growing depending on the player's height/width. I say two separate options since mods like "More Player Models" do the line of sight change but not the hit-box change due to the need to be used on Vanilla servers. Since MCA-Reborn does require being on both client and server, this shouldn't be an issue. Another reason they're separate options is for servers/players who may want to keep everyone's hit-box the same (since that CAN affect gameplay) but allow line of sight changes since that'd be purely cosmetic.

**Describe alternatives you've considered**
Due to the amount of work that might go into the hit-box part, I'm perfectly okay with just the line of sight part getting added.

**Additional context**
N/A

**Comment by TheMCSpacePro** [comment]
Just thought of something else: please make it so that one can choose gender and height factor independently from one-another. In layman's terms: the ability to make a girl use the height scale of a boy, and vice versa. This should be as simple as a button toggle in the "Body" tab with three options: "Auto" (current system), "Masculine" (use the male height slider regardless of gender), and "Feminine" (use the female height slider regardless of gender).

**Comment by Luke100000** [comment]
Changing the players bounding box is WIP and scheduled to be added with the reborn update. For now it is disabled as it conflicts with many PvP or player model affecting cases.

I'm not sure whats the problem with line of sight, as that's already added. You need to shift once on older versions to force a resync.

Gender only has a 5% influence on height to simulate the human height average. As long as you don't go with ridiculous values it has no influence on customization.

**Comment by BassRemedy** [comment]
just hoping this hitbox thing actually gets added at some point, its quite annoying that you cant actually play as a shorter character and fit in smaller places, or if a friend tries to shoot an arrow over your head it hits you because the hitbox does not change
